server.js

```diff
@@ -83,4 +83,23 @@ app.get("/books/", async (request, response) => {
   response.send(books);
 });
 
+app.put("/books/:id/:email", async (request, response) => {
+  const id = request.params.id;
+  const email = request.params.email;
+  console.log(id);
+  try {
+    const bookToUpdate = await Book.findOne({ _id: id, email: email });
+    if (!bookToUpdate) {
+      response.status(400).send("Unable to update book");
+      return;
+    }
+    const updatedBook = await Book.findByIdAndUpdate(id, request.body, {
+      new: true,
+    });
+    response.send(updatedBook);
```

Comment:
Nice work so far on this, good progress!

I noticed here that you want to res.status.(200).send(updatedBook) or res.status.(204).send(updatedBook)